tools/clang/test/HLSLFileCheck/hlsl/semantics/legacy/D3D9_effect_ps_6_0Tex2D.hlsl

```diff
@@ -0,0 +1,27 @@
+// RUN: %dxc /E main_fragment -T ps_6_0 %s | FileCheck %s
```

Comment:
I think these three files could be merged into one that tests all of these possibilities. That would make maintaining them simpler.

You could either include a define which specifies the type of the "decal" parameter and specify it on the command line using something like:
```
// RUN: %dxc -D PARAMTYPE=Texture2D<float4> ...
// RUN: %dxc -D PARAMTYPE=SamplerState ...
...
int main_fragment(float2 texCoord : TEXCOORD0, uniform PARAMTYPE decal: TEXUNIT0
...
```
or you could just have an entry function that includes all of the invalid parameters provided the compiler produces errors for each one.